xonsh/environ.py

```diff
@@ -891,6 +912,7 @@ def current_branch(pad=True):
         branch = get_hg_branch()
     if isinstance(branch, subprocess.TimeoutExpired):
         branch = '<branch-timeout>'
+        _first_branch_timeout_message()
```

Comment:
Okay, sounds good.  Yeah, if we had that, then I would feel a little bit better.  So maybe I'll go ahead and merge this in, and then I can try to get the `{color_and_branch}` in in the near future (before the next release).